cvrf2csaf/section_handlers/vulnerability.py

```diff
@@ -91,6 +103,63 @@ def _handle_remediations(root_element):
 
         return remediations
 
+    @staticmethod
+    def _parse_score_set(score_set_element, mapping, version, json_property, product_status):
+        """Parses ScoreSetV2 or ScoreSetV3 element."""
+
+        cvss_score = {'version': version}
+
+        for cvrf, csaf in mapping.items():
+            if hasattr(score_set_element, cvrf):
+                cvss_score[csaf] = getattr(score_set_element, cvrf).text
+
+        # TODO: How to handle missing products entries?
+        product_ids = []
+        if hasattr(score_set_element, 'ProductID'):
+            product_ids = [product_id.text for product_id in score_set_element.ProductID]
+        elif product_status:
+            product_ids = []
+            if product_status.get('known_affected'):
+                product_ids.extend(product_status['known_affected'])
+            if product_status.get('first_affected'):
+                product_ids.extend(product_status['first_affected'])
+            if product_status.get('last_affected'):
+                product_ids.extend(product_status['last_affected'])
+
+            product_ids = list(set(product_ids))
+
+        if 'baseScore' not in cvss_score:
+            logging.error('No CVSS base score found on the input!')
+
+        if 'vectorString' not in cvss_score:
+            logging.error('No CVSS vector string found on the input!')
+
+        if len(product_ids) == 0:
+            logging.error('No product_id entry for CVSS score set!')
+
+        score = {json_property: cvss_score,
+                 'products': product_ids}
+
+        return score
+
+    def _handle_scores(self, root_element, product_status):
+        scores = []
+
+        # TODO: VectorV3 is not mandatory on input, but vectorString is mandatory on output
+        # TODO: hardcoded cvss version 3.0/3.1 or config param?
+
+        if hasattr(root_element, 'ScoreSetV2'):
```

Comment:
In case we want more focus on the data we could refactor as:
```python
        score_variants = (
            ('ScoreSetV2', self.cvss_v2_mapping, '2.0', 'cvss_v2'),
            ('ScoreSetV3', self.cvss_v3_mapping, '3.0', 'cvss_v3'),
        )
        for score_variant, mapping, score_version, target in score_variants:
            for score_set in getattr(root_element, score_variant, []):
                 scores.append(self._parse_score_set(score_set, mapping, score_version, target, product_status))
```
(not tested with lxml etrees ...)